File: silver/18110/python/main.py
import sys
input = sys.stdin.readline

if (__name__ == "__main__"):
    n = int(input())

    if (n == 0): print(0); exit(0)

    array = list(int(input()) for _ in range(n))

    array.sort()

    cut = (n * 15 + 50) // 100
    trimmed = n - cut * 2

    if (trimmed <= 0): print(0); exit(0)
    
    sum = 0
    trimmed_arr = array[cut:] if cut == 0 else array[cut:-cut]
    for data in trimmed_arr:
        sum += data
    print((2 * sum + trimmed) // (2 * trimmed))

File: silver/18110/c++/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

int main()
{
    int n;
    std::cin >> n;

    if (n == 0) {
        std::cout << 0;
        return 0;
    }

    std::vector<int> array(n);
    for (int i = 0; i < n; i++) {
        std::cin >> array[i];
    }

    std::sort(array.begin(), array.end());

    int cut = (n * 15 + 50) / 100;
    int trimmed = n - cut * 2;

    if (trimmed <= 0) {
        std::cout << 0;
        return 0;
    }

    int sum = 0;
    for (int i = cut; i < n - cut; i++) {
        sum += array[i];
    }

    std::cout << (2 * sum + trimmed) / (2 * trimmed);

    return 0;
}